packages/next-swc/crates/core/src/emotion/hash.rs

```diff
@@ -0,0 +1,62 @@
+// Ported from https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash2.cpp#L37-L86
```

Comment:
Is it necessary to add any license info?

Reply:
From the [README](https://github.com/aappleby/smhasher) of `SMHasher`:

> SMHasher is released under the MIT license. All MurmurHash versions are public domain software, and the author disclaims all copyright to their code.

I think the reference comments are enough.